cmd/assetsvc/handler.go

```diff
@@ -336,3 +379,25 @@ func newChartVersionListResponse(c *models.Chart) apiListResponse {
 
 	return cvl
 }
+
+func decodeParam(param string, w http.ResponseWriter) string {
+	decodedParam, err := url.PathUnescape(param)
+	if err != nil {
+		log.WithError(err).Errorf("could not decode param %s", decodedParam)
+		if w != nil {
+			response.NewErrorResponse(http.StatusBadRequest, "could not decode param: "+param).Write(w)
+		}
```

Comment:
Thanks for the improvement. You're totally right; I forgot adding the `return` so my error handling was being completely useless... 
It's a bit tedious, perhaps adding a middleware/filter/whatever it is called in go for handling this could be interesting.
I will address this change following the more idiomatic approach.

This is also related to the management of the routes: it's really annoying to have to define a apiv1.Methods("GET").Path.(...).Queries(...)` per each combination of path/params instead of having one per path.

Reply:
> This is also related to the management of the routes: it's really annoying to have to define a apiv1.Methods("GET").Path.(...).Queries(...)` per each combination of path/params instead of having one per path.

+100 . For a separate branch, but yes, I've wanted to see if we can just have one route regardless of params - not sure whether it's a result of the lib being used or our use of that lib.